android/core/src/main/java/com/mongodb/stitch/android/core/auth/StitchAuth.java

```diff
@@ -20,20 +20,33 @@
 import com.mongodb.stitch.android.core.auth.providers.internal.AuthProviderClientFactory;
 import com.mongodb.stitch.android.core.auth.providers.internal.NamedAuthProviderClientFactory;
 import com.mongodb.stitch.core.auth.StitchCredential;
+import com.mongodb.stitch.core.auth.internal.StitchAuthRequestClient;
+import com.mongodb.stitch.core.internal.net.StitchRequestClient;
 
 import java.io.Closeable;
 
 public interface StitchAuth extends Closeable {
+  /**
+   * Gets an authenticated client for the given authentication provider. Most authentication
+   * providers will allow creation of a client without a name of the provider.
+   *
+   * @param provider The provider that will create a client for the authentication provider.
+   * @param <ClientT> The type of client to be returned by the provider.
+   * @return A client to interact with the authentication provider.
+   */
+  <ClientT> ClientT getAuthenticatedProviderClient(
```

Comment:
I'd really like to avoid having an extra method like this with this kind of name. There should be another way to do this. If the type erasure makes it impossible or just the lack of the `where` constraints, can we just make the factories accept both a requestclient and authrequestclient. If it goes unused it's fine. The users of this SDK won't care.

Reply:
I'm going with the factory accepting both types of request clients. I could not figure out another way to do it given the type erasure.